### PR TITLE
Basic support for showing logged in user info the the top nav

### PR DESCRIPTION
**Pull request by TomNUSDS**
- User initials. 
- User details on click
- User details popup is an inline quick impl. Things like escape key don't close it.
- IMPORTANT: Logout stubbed out. Need to get details for Okta logout since there seems to be several ways to do it. Right now it just alerts.
- User details also shown in "mobile" side-menu view.
- Minor fixes to layout:
  - close boxes from Test Queue tab had z-index so high it would overlap mobile menu.
  - logo in header was so big it was forcing the whole top nave to be taller than it should be.
  - Used consistent close box for topnav (but it's visibly different from close box for Test Queue delete, which is probably good).

Demo here: https://youtu.be/TUu7LHpiQTs

TODO: 
- Implement actual Okta logout. Probably just a redirect.
- Need to test things with username like if there's not middle name or if there is a suffix like APRN

### CODE REVIEW

File: frontend/src/app/commonComponents/Header.jsx
Comment (dmethvin-gov) [inline frontend/src/app/commonComponents/Header.jsx:86]: This should stay a button for a11y/508 reasons. You can style it to look however you need though.

File: frontend/src/app/commonComponents/Header.jsx
Comment (dmethvin-gov) [inline frontend/src/app/commonComponents/Header.jsx:62]: If it's okay to not have logout for MVP we could ship after removing these two lines. Another half-option would be to just redirect the user to the home page without logging them out.

File: frontend/src/app/commonComponents/Header.jsx
Comment (TomNUSDS) [reply frontend/src/app/commonComponents/Header.jsx]: Agreed. Fixed in both places.

File: frontend/src/app/commonComponents/Header.jsx
Comment (TomNUSDS) [reply frontend/src/app/commonComponents/Header.jsx:62]: I'm waiting to hear back from @nickrobison-usds about the link to use. I think @TimBest wants to hold off on the commit until it's in the change.

Comment (timothybest-usds) [reply frontend/src/app/commonComponents/Header.jsx:62]: Happy to ship the changes with the logout as `display:none` it you want

Comment (TomNUSDS) [reply frontend/src/app/commonComponents/Header.jsx:62]: Let's talk about this at standup? I feel like this whole feature is more about getting usability feedback from users than actual functionality at this stage.

**Review by timothybest-usds — APPROVED** [review]

**Comment by timothybest-usds** [comment]
I pushed the logout link (currently) navigating to it returns a 404. I suspect #225 needs to me merged first

Also I think @hmyers-cms should review the design as it is quite different then the mock in https://github.com/CDCgov/prime-central/issues/135

screen grabs from my local env
Screen Shot 2020-12-03 at 4 14 39 PM
Screen Shot 2020-12-03 at 4 14 31 PM
Screen Shot 2020-12-03 at 4 14 26 PM

**Comment by hmyers-cms** [comment]
Few things don't match the design here:
- We shouldn't show user initials next to the person icon
- The style of the dropdown should match the style on the search dropdown in the test queue (rounded corners, drop shadow)
- No need for an X button, it should just close by clicking anywhere outside of it
- Everything appears right-aligned but should be left-aligned
- The background color in the nav should turn navy, not gray, to match the active state for the other nav links
- We should be showing the user's email address and the name of the organization under their name in the dropdown
- The log out button should just be a simple link style—we aren't using red buttons anywhere else (however if we were, it's worth noting that it doesn't look like you're using USWDS button styles there, just flagging for future reference)

We didn't design for the mobile slide-out menu but the way you've set it up is fine so long as the styles match, and where you have the initials now I'd have it say "User Profile" (similar to how you added the word "Settings" next to that icon)

Design for reference (also can be found in the ticket that Tim linked above)
Screen Shot 2020-12-03 at 4 37 47 PM

**Comment by TomNUSDS** [comment]
@TimBest There might be some stylesheet issues after the merge (for example the close box should not be red).

Some of the UI differences were just minor corrections to get it done with less code. For example, I we currently don't have "popup" UI that isn't a **full** modal anywhere else. So, I coded one quickly inline. Since it didn't handle clicking outside the box or pressing "escape" to dismiss, I put in the close box.

The side menu look wasn't defined anywhere that I could see, so I guessed.

I'd already coded up the user initials before I heard back hmyers-cms (time zone thing)  She said they weren't and I can removed them.

**Comment by TomNUSDS** [comment]
@hmyers-cms  I'll make these changes and investigate CSS issues.

Just some things to consider for the next UI pass.
 - email address can leak PII between users (e.g. staff are sharing a tablet). They are also tend to be long and don't wrap well.
 - I'll look into removing the close box. @dmethvin-gov @nickrobison-usds do you recommend some non-modal way of closing this "drop down menu like div"?

**Comment by TomNUSDS** [comment]
@hmyers-cms @timothybest-usds 
Screen Shot 2020-12-03 at 3 52 06 PM

Unfortunately, email address not currently in the api, so it can't appear in the popup. It's easy enough to add once it's available.

**Comment by hmyers-cms** [comment]
Couple things still missing for the styles:
- The style of the dropdown should match the style on the search dropdown in the test queue (rounded corners, drop shadow)
- The background color in the nav should turn navy to match the active state for the other nav links
- The link style doesn't look like it matches other links on the site (font weight/underline)

**Comment by benwarfield-usds** [comment]
I believe address is sneakily in the API as the username, but I'll check.

**Comment by TomNUSDS** [comment]
Oh, username kind of makes sense. But I'm dubious we want/need to show the sign-in email since it can be seen by anyone who picks up a tablet.

**Comment by benwarfield-usds** [comment]
Created #234 to track the API update--let's put any further discussions/decisions about the future state there, since this issue is officially done for now.

**Comment by hmyers-cms** [comment]
I'm not concerned about showing the email address. IF they share a tablet and hand off without logging out, then we have much bigger issues than having an email address showing. Also they're almost certainly using a work email and handing it off to a coworker. But right now the users we have aren't sharing devices anyways and we'll continue thinking about and improving the design going forward

**Comment by TomNUSDS** [comment]
Fair point. I tend to fall on the side of showing "least data" where possible. Thanks for your continued deep thinking on such things and we should use #234 going forward.

**Pull request merged**
